packages/contracts-bedrock/src/Safe/LivenessModule.sol

```diff
@@ -0,0 +1,216 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Safe, OwnerManager } from "safe-contracts/Safe.sol";
+import { Enum } from "safe-contracts/common/Enum.sol";
+import { OwnerManager } from "safe-contracts/base/OwnerManager.sol";
+import { LivenessGuard } from "src/Safe/LivenessGuard.sol";
+import { ISemver } from "src/universal/ISemver.sol";
+
+// TODO(maurelian): remove me
+import { console2 as console } from "forge-std/console2.sol";
+
+/// @title LivenessModule
+/// @notice This module is intended to be used in conjunction with the LivenessGuard. In the event
+///         that an owner of the safe is not recorded by the guard during the liveness interval,
+///         the owner will be considered inactive and will be removed from the list of owners.
+///         If the number of owners falls below the minimum number of owners, the ownership of the
+///         safe will be transferred to the fallback owner.
+contract LivenessModule is ISemver {
+    /// @notice The Safe contract instance
+    Safe internal immutable SAFE;
+
+    /// @notice The LivenessGuard contract instance
+    ///         This can be updated by replacing with a new module and switching out the guard.
+    LivenessGuard internal immutable LIVENESS_GUARD;
+
+    /// @notice The interval, in seconds, during which an owner must have demonstrated liveness
+    ///         This can be updated by replacing with a new module.
+    uint256 internal immutable LIVENESS_INTERVAL;
+
+    /// @notice The minimum number of owners before ownership of the safe is transferred to the fallback owner.
+    ///         This can be updated by replacing with a new module.
+    uint256 internal immutable MIN_OWNERS;
+
+    /// @notice The fallback owner of the Safe
+    ///         This can be updated by replacing with a new module.
+    address internal immutable FALLBACK_OWNER;
+
+    /// @notice The storage slot used in the safe to store the guard address
+    ///         keccak256("guard_manager.guard.address")
+    uint256 internal constant GUARD_STORAGE_SLOT = 0x4a204f620c8c5ccdca3fd54d003badd85ba500436a431f0cbda4f558c93c34c8;
+
+    /// @notice The address of the first owner in the linked list of owners
+    address internal constant SENTINEL_OWNERS = address(0x1);
+
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.0
+    string public constant version = "1.0.0";
+
+    // Constructor to initialize the Safe and baseModule instances
+    constructor(
+        Safe _safe,
+        LivenessGuard _livenessGuard,
+        uint256 _livenessInterval,
+        uint256 _minOwners,
+        address _fallbackOwner
+    ) {
+        SAFE = _safe;
+        LIVENESS_GUARD = _livenessGuard;
+        LIVENESS_INTERVAL = _livenessInterval;
+        MIN_OWNERS = _minOwners;
+        FALLBACK_OWNER = _fallbackOwner;
+    }
```

Comment:
Any sanity checks needed on verifying this address? Maybe `require(_fallbackOwner.code.length > 0)` is a good sanity check? Alternatively, if we typically prefer to keep all these checks off-chain in scripts, that's ok with me

Reply:
This is a good callout, but I generally prefer to avoid these validations. They make testing harder and we are definitely going to implement config checks prior to deploying this.